curriculum/challenges/english/25-front-end-development/lab-contact-form/66f26c32ec6f90df01a44f60.md

```diff
@@ -0,0 +1,420 @@
+assert.isNotEmpty(el?.getAttribute('name'));
+```
+
+Within the `form`, you should have a `textarea` with an `id`, `name`, and a `required` attribute.
+
+```js
+const el = document.querySelector('.form-container form textarea');
+assert(!!el && el.tagName === 'TEXTAREA');
+assert(!!el && el.required);
+assert.isNotEmpty(el?.getAttribute('id'));
+assert.isNotEmpty(el?.getAttribute('name'));
+```
+
+Within the `form`, you should have a `button` element with a `type` attribute of `submit`.
+
+```js
+const el = document.querySelector('.form-container form button');
+assert(!!el && el.tagName === 'BUTTON');
+assert(!!el && el.getAttribute('type') === 'submit');
+```
+
+Your button should have a text of `"Submit"`.
+
+```js
+const el = document.querySelector('.form-container form button');
+assert(!!el && el.innerText === 'Submit');
+```
+
+You should have a `label` element for the `#name` `input` with a `for` attribute matching the `id` value on the corresponding `input`.
+
+```js
+const nameInput = document.querySelector('.form-container form input[type="text"]')?.getAttribute('id');
+
+const labels = document.querySelectorAll('.form-container form label');
+
+assert(labels.length > 0);
+
+let found = false;
+
+for (let label of labels) {
+    if (label.getAttribute('for') === nameInput) {
+        found = true;
+        break;
+    }
+}
+
+assert(found);
+```
+
+You should have a `label` element for the `#email` `input` with a `for` attribute matching the `id` value on the corresponding `input`.
+
+```js
+const emailInput = document.querySelector('.form-container form input[type="email"]')?.getAttribute('id');
+
+const labels = document.querySelectorAll('.form-container form label');
+
+assert(labels.length > 0);
+
+let found = false;
+
+for (let label of labels) {
+    if (label.getAttribute('for') === emailInput) {
+        found = true;
+        break;
+    }
+}
+
+assert(found);
+```
+
+You should have a `label` element for the `#message` `textarea` with a `for` attribute matching the `id` value on the `textarea`.
+
+```js
+const messageInput = document.querySelector('.form-container form textarea')?.getAttribute('id');
+
+const labels = document.querySelectorAll('.form-container form label');
+
+assert(labels.length > 0);
+
+let found = false;
+
+for (let label of labels) {
+    if (label.getAttribute('for') === messageInput) {
+        found = true;
+        break;
+    }
+}
+
+assert(found);
+```
+
+The labels shouldn't be empty.
+
+```js
+const labels = document.querySelectorAll('.form-container form label');
+
+assert(labels.length > 0);
+
+for (let label of labels) {
+    const labelText = label.textContent.trim();
+    assert(labelText.length > 0);
+}
+
+```
+
+The `form-container` `div` should have a background color.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.form-container')?.backgroundColor);
+```
+
+The `form-container` `div` should have a visible border that is also rounded.
```

Comment:
```suggestion
The `form-container` `div` should have a `border-radius`.
```

Not sure on this one. It doesn't say anywhere to add a border. Maybe you meant that it should have a "visible border" in that, it should have a `background-color`, which makes it appear to have a border. We don't have to go with my suggestion.

Reply:
I added an outline as well, but that doesn't match the demo. So I have removed it.